File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int,char> map;

        for(auto x:nums) map[x]='u';
        int length = 0;

        for(auto x:nums){
            if(map[x] == 'v' || map[x-1]) continue;
            int temp=0;
            int k = x;
            while(map[k]){
                map[k] = 'v';
                temp++; 
                k++;
            }
            length = temp > length ? temp : length;
        }

        return length;
    }
};

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-3.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} head
     * @param {number} n
     * @return {ListNode}
     */
    removeNthFromEnd(head, n) {
        let node = head;
        let temp = null;
        while(node.next){
            n--;
            node = node.next;
            if(n==0) temp = head;
            else if(n<0) temp = temp.next;
        }
        console.log(temp);
        if(!temp) head = head.next;
        else temp.next = temp?.next?.next || null;
        return head;
    }
}
